platforms/chibios/bootloaders/sn32_dfu.c

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2022 QMK
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "bootloader.h"
+
+#include <ch.h>
+#include <hal.h>
+
+#       define SYMVAL(sym) (uint32_t)(((uint8_t *)&(sym)) - ((uint8_t *)0))
+extern uint32_t __ram0_end__;
+#       define BOOTLOADER_MAGIC 0xDEADBEEF
+#       define MAGIC_ADDR (unsigned long *)(SYMVAL(__ram0_end__) - 4)
+
+__attribute__((weak)) void bootloader_jump(void) {
+    *MAGIC_ADDR = BOOTLOADER_MAGIC;  // set magic flag => reset handler will jump into boot loader
+    // Wait for memory to be set before the reset
+    // FIXME: update to wait funcion
+    for(volatile uint32_t i = 0; i < 32; i++){
+        __NOP();
+    }
```

Comment:
Should use the `wait_us()` / `wait_ms()` function right?

Reply:
probably, we just now got the wait_ functions working reliably. Still not sure how to translate that to us value

Reply:
1us seems good enough. updated